sql/postgres/migrate.go

```diff
@@ -208,20 +208,23 @@ func (s *state) modifyTable(ctx context.Context, modify *schema.ModifyTable) err
 			from, ok1 := change.From.Type.Type.(*schema.EnumType)
 			to, ok2 := change.To.Type.Type.(*schema.EnumType)
 			switch {
-			// Enum was added.
-			case !ok1 && ok2:
-				if err := s.addTypes(ctx, change.To); err != nil {
-					return err
-				}
 			// Enum was changed.
 			case ok1 && ok2 && from.T == to.T:
 				if err := s.alterType(from, to); err != nil {
 					return err
 				}
-			// Not an enum, or was dropped.
-			default:
-				changes = append(changes, change)
+				// If only the enum values were changed,
+				// there is no need to ALTER the table.
+				if change.Change == schema.ChangeType {
```

Comment:
I am confused. There is no need to alter the table but it is altered just above this block

Reply:
If only the enum was changed (e.g. add values), there's no need to modify the column (on ALTER TABLE). See tests, or let me know if it needs to phrasing work.